bundles/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/accessories/HomekitValveImpl.java

```diff
@@ -119,19 +119,13 @@ private void startTimer() {
         int duration = getDuration();
         logger.trace("start timer for duration {}", duration);
         if (duration > 0) {
-            ScheduledFuture<?> future = valveTimer;
-            if (future != null && !future.isDone()) {
-                future.cancel(true);
-            }
+            stopTimer();
             valveTimer = timerService.schedule(() -> {
                 logger.trace("valve timer is over. switching off the valve");
                 switchOffValve();
                 // let home app refresh the remaining duration, which is 0
                 ((GenericItem) getRootAccessory().getItem()).send(RefreshType.REFRESH);
             }, duration, TimeUnit.SECONDS);
-
-            // let home app refresh the remaining duration, which is 0
-            ((GenericItem) getRootAccessory().getItem()).send(RefreshType.REFRESH);
```

Comment:
Wouldn't you still need this since you removed it from `stopTimer`?

Reply:
i have added to the caller method, setValveActive. it will call REFRESH in both cases - start and stop timer. 
```
 if (homekitTimer) {
                if ((state == ActiveEnum.ACTIVE)) {
                    startTimer();
                } else {
                    stopTimer();
                }
                // let home app refresh the remaining duration
                ((GenericItem) getRootAccessory().getItem()).send(RefreshType.REFRESH);
```

Reply:
Ah, didn't notice that. Looks good to me then.